### ISSUE TITLE
IPFS mirror

### ISSUE THREAD

**Issue by RubenKelevra** [endeavouros-team/mirrors#1]
Hey guys,

I'm running an ArchLinux packages IPFS-Cluster under the domain [pacman.store](http://pacman.store), which allows everyone to join by running a small daemon next to the IPFS node. This is called [collaborative cluster](https://collab.ipfscluster.io/).

I was wondering if you're interested in having me fetch your packages as well and add them to the cluster. This would allow people to fetch your ISOs and packages via an IPFS node, which would then continue to seed the files - like bittorrent.

**Comment by RubenKelevra** [comment]
I've added the current iso to my main server as an example.

/ipfs/bafykbzacedyo7klhyzgzekwpxvuxrqihsl4rcgzpvxcrptyoqf43ruio3buis

**Comment by RubenKelevra** [comment]
@AlpixTM @killajoe :)

**Comment by AlpixTM** [comment]
Hey Ruben,

sorry for our late reply and thank you for your offer and idea.

Right now we think that we don't need that. We might consider using IPFS in
the future.


Greetings Lukas


Am Di., 27. Juli 2021 um 15:15 Uhr schrieb @RubenKelevra <
***@***.***>:

**Comment by RubenKelevra** [comment]
@AlpixTM I understand, that you don't need it - but that was never in question.

I would just like to offer it as an alternative to regular HTTPS/HTTP mirrors. I just need rsync access to one of your tier1 or main mirrors to make that happen. :)

**Comment by AlpixTM** [comment]
Hey Ruben,

we only offer tier1 access to our official mirrors, but you can sync from an official mirror.

E.g. you could sync from mirror.erickochen.nl::endeavouros 

Greetings Lukas

**Comment by killajoe** [comment]
reopen if you need, but looks solved to me ?

**Comment by RubenKelevra** [comment]
Yeah, will set it up as soon as I find time :)

**Comment by RubenKelevra** [comment]
@AlpixTM @killajoe just noticed that there are ".sha512sum.1" files in the iso folder, are they supposed to be there?

And there are a lot of old iso releases ... do we really need to hold them on the mirrors? :)

**Comment by RubenKelevra** [comment]
I've added it to the cluster. It can be viewed in the Browser via

http://endeavouros.pkg.pacman.store/

For pacman you just have to configure it to use the local ipfs node, which should look like this:

`Server = http://endeavouros.pkg.pacman.store.ipns.localhost:8080/repo/$repo`

**Comment by AlpixTM** [comment]
> @AlpixTM @killajoe just noticed that there are ".sha512sum.1" files in the iso folder, are they supposed to be there?

No, thanks for the information, I will remove it.
> And there are a lot of old iso releases ... do we really need to hold them on the mirrors? :)

No, it's planned to remove the old ones soon

**Comment by AlpixTM** [comment]
> I've added it to the cluster. It can be viewed in the Browser via
> 
> http://endeavouros.pkg.pacman.store/
> 
> For pacman you just have to configure it to use the local ipfs node, which should look like this:
> 
> `Server = http://endeavouros.pkg.pacman.store.ipns.localhost:8080/repo/endeavouros/x86_64`

Would that work for everybody like a normal http mirror, or is some config with pacman needed?
Does a user need to run an IPNS instance to connect?

**Comment by RubenKelevra** [comment]
@AlpixTM there's no special config needed for pacman. The decentralized nature of IPFS just means that there are some operations that have more latency. That's why I recommend having the mirror 3 times in the list, to have pacman retry it.

While you can access the IPFS network via publicly available gateway you loose a lot of features doing this (and might run into some rate-limiting).

It's recommended to install ipfs on each machine which should use the mirror or have one centralized ipfs instance in a local network since the packages will be cached and offered to other computers in the local network at LAN speed.

Additionally, each node that downloaded a package via the mirror will hold it for a while in its cache (until the cache gets wiped the next time). So there are many sources available if a package is often installed - which should allow for organic growth instead of having a lot of servers sitting around.

So in short the ipfs daemon will replace the lan sharing scripts/addons for pacman, while there's zero configuration necessary, even if you switch between different networks and allow for a more organic decentralized growth of the package exchange infrastructure.

**Comment by RubenKelevra** [comment]
I've updated the install instructions to include Endeavouros

https://github.com/RubenKelevra/pacman.store#use-the-pkg-cache-with-pacman

**Comment by RubenKelevra** [comment]
@AlpixTM wrote
> Hey Ruben,
> 
> we only offer tier1 access to our official mirrors, but you can sync from an official mirror.
> 
> E.g. you could sync from mirror.erickochen.nl::endeavouros 

Hey Lukas,

this mirror is gone:

```
$ nslookup mirror.erickochen.nl 1.1.1.1
Server:		1.1.1.1
Address:	1.1.1.1#53

** server can't find mirror.erickochen.nl: NXDOMAIN
```

Is there an alternative I can tap into?

**Comment by AlpixTM** [comment]
Hey,

you can use rsync://mirror.jingk.ai/endeavouros/

Does that work for you?

Greetings Lukas

**Comment by RubenKelevra** [comment]
Hey @AlpixTM,

thanks!

I've updated my config, if it's not working I let you know :)

https://github.com/RubenKelevra/rsync2ipfs-cluster/compare/075b6c349fbe3c193604445cf40fd74b59ca9b76...master

**Comment by RubenKelevra** [comment]
Hey @AlpixTM,

while this server works it's kind of slow. I feel like I might overload it when I leech from it all the time.

Current download speed is just 16 Mbit/s.

Is there any fast server available? :)

**Comment by RubenKelevra** [comment]
@AlpixTM this one is fine, getting around 250 MByte/s  :)

Thanks again!
